CMakeLists.txt

```diff
@@ -41,6 +41,17 @@ if(FIPS)
   endif()
 endif()
 
+if(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+  # OpenBSD by defaults links with --execute-only this is problematic for two reasons:
+  # 1. The FIPS shared and static builds need to compute the module signature hash by reading the .text section
+  # 2. s2n-bignum x86 assembly sources still have references to static data in the .text section
+  if(NOT BUILD_SHARED_LIBS)
```

Comment:
a'ha you disable it... This would be very nice to have

Reply:
Yeah need to poke that ticket we have with them to determine if there is a technical block for it (maybe related to formal verification?), or if we can just submit a PR for them. But I don't want to block this on that for the time being.

Regardless for FIPS we don't have a good strategy without disabling this, as the premise of our integrity check is to compute the hash of the rodata and text section. With text being execute-only and no read that doesn't work.